### PR TITLE
Use the same processors for all currently supported provisioning classes

### PR DESCRIPTION
**Pull request by aleksandra-malinowska**
#### What this PR does / why we need it:

Refactor that removes CombinedProcessor (we don't need separate processors for any currently envisioned provisioning classes) and adds a source-of-truth for supported provisioning classes.

The only change in behavior is including best-effort-atomic provisioning class as supported in now generic provisioning requests processor.

#### Which issue(s) this PR fixes:

Issue: #6815

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/proposals/provisioning-request.md
```

/cc @yaroslava-serdiuk

### CODE REVIEW

**Comment by aleksandra-malinowska** [comment]
@yaroslava-serdiuk please confirm if lack of validation in injector was intentional, I'll add it if it was a bug.

**Comment by yaroslava-serdiuk** [comment]
Thanks for the changes!
Let's update the main as well: https://github.com/kubernetes/autoscaler/blob/c61e295cb233dfbe3be4f34138618783aff03cc1/cluster-autoscaler/main.go#L510

Otherwise /lgtm

**Comment by yaroslava-serdiuk** [comment]
> Yaroslava Serdiuk please confirm if lack of validation in injector was intentional, I'll add it if it was a bug.

Sorry for late reply, I just realised that this is a bug and we actually should add validation in injector. 
If it's ok for you, can you add it in this PR? Otherwise I can add it later in separate PR.

**Comment by aleksandra-malinowska** [comment]
>Sorry for late reply, I just realised that this is a bug and we actually should add validation in injector.
>If it's ok for you, can you add it in this PR? Otherwise I can add it later in separate PR.

Let's merge this and add it later (preferably using the map introduced in this PR as source-of-truth).

File: cluster-autoscaler/processors/provreq/provisioning_request_processor.go [hunk 1]
```diff
@@ -17,41 +17,96 @@ limitations under the License.
...
-		p.Process(provReqs)
+
+	p.Process(provReqs)
+}
+
+// Process iterates over ProvisioningRequests and apply:
+// -BookingExpired condition for Provisioned ProvisioningRequest if capacity reservation time is expired.
+// -Failed condition for ProvisioningRequest that were not provisioned during defaultExpirationTime.
+// TODO(yaroslava): fetch reservation and expiration time from ProvisioningRequest
+func (p *provReqProcessor) Process(provReqs []*provreqwrapper.ProvisioningRequest) {
```
[the remark this answers is not in the document]
Comment (x13n) [reply cluster-autoscaler/processors/provreq/provisioning_request_processor.go:67]: Yeah, I noticed, ended up copying both files and running a diff locally :)

**Pull request merged**